unityfiles/Assets/Scripts/Helper/Letter_gen_Finn_pos/Sudo_random_letter_generator.cs

```diff
@@ -0,0 +1,211 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using UnityEngine;
+
+
+
+public class Sudo_random_letter_generator : MonoBehaviour
+{
+
+    // -- Config
+    private int reduceWordSizeMin = 2;
+    private int reduceWordSizeMax = 6;
+    private int startWordSertchAt = 2;
+
+
+
+
+    private SettValuePairs allSVP;
+    private LetterFreq letterFreq;
+    void Start()
+    {
+       using (StreamReader r = new StreamReader("Assets/Scripts/Helper/Letter_gen_Finn_pos/letter_sett_lists/letter_frequensy_weight_100.json"))
+        {
+            string jsonText = r.ReadToEnd();
+            this.letterFreq = JsonUtility.FromJson<LetterFreq>(jsonText);
+
+        }
+        
+        using (StreamReader r = new StreamReader("Assets/Scripts/Helper/Letter_gen_Finn_pos/letter_sett_lists/score_sorted_sett_size_dict.json"))
+        {
+            string jsonText = r.ReadToEnd();
+            this.allSVP = JsonUtility.FromJson<SettValuePairs>(jsonText);
+        }
+
+
+        float t1 = System.DateTime.Now.Millisecond;
+        for (int i = 0; i < 100; i++)
+        {
+            print(this.GenerateLetter());
+        }
+        float rt = System.DateTime.Now.Millisecond - t1;
+        print("avg run time per gen in ms=" + rt/100);
+        
+    }
+
+    // --utillity funticions (aka shuld be in standard lib)
+    List<string> RemoveAllOfAFromB(List<string> A, List<string> B)
+    {
+        List<string> ret = new List<string>(B);
+
+        foreach (string item in A)
+        {
+            ret.Remove(item);
+        }
+        return ret;  
+    }
+
+
+    // cheks if a contains all elements in b
+    bool IsAsubsettB(List<String> A, List<String> B)
```

Comment:
IsSubsetB

Reply:
is A an subset of B. actually not a typo